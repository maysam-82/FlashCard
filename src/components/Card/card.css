.card {
	cursor: pointer;
	font-size: 1rem;
}

.card-prompt {
	width: calc(50% - 1px);
}

.card-answer {
	width: calc(50% - 1px);
}

.text-hidden {
	visibility: hidden;
}

.text-revealed {
	animation-name: fade-in;
	animation-duration: 1s;
	visibility: visible;
}

@keyframes fade-in {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}
