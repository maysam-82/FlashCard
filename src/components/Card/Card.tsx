import React from 'react';
import './card.css';
export interface IProps {
	prompt: string;
	answer: string;
}
export interface IState {
	isAnswerHidden: boolean;
}

class Card extends React.Component<IProps, IState> {
	constructor(props: IProps) {
		super(props);

		this.state = {
			isAnswerHidden: false,
		};
	}

	render() {
		const { answer, prompt } = this.props;
		const { isAnswerHidden } = this.state;
		return (
			<div
				className="card border rounded m-2 p-2 d-flex  flex-row justify-content-between align-items-center"
				onClick={() => this.setState({ isAnswerHidden: true })}
			>
				<div className="card-prompt text-left">
					<h4 className="m-0">{prompt}</h4>
				</div>
				<div className="card-answer text-right">
					<h4
						className={`m-0 ${
							isAnswerHidden ? 'text-revealed' : 'text-hidden'
						}`}
					>
						{answer}
					</h4>
				</div>
			</div>
		);
	}
}
export default Card;
