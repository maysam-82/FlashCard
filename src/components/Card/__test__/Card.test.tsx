import React from 'react';
import Card, { IProps, IState } from '../Card';
import { shallow } from 'enzyme';

describe('Card component', () => {
	const component = shallow<Card, IProps, IState>(
		<Card answer="test answer" prompt="test prompt" />
	);
	it('should render Card correclty', () => {
		expect(component).toMatchSnapshot();
	});

	describe('Before user clicks on `Card`', () => {
		it('should not show the answer', () => {
			expect(component.find('.text-hidden').exists()).toBeTruthy();
		});
	});

	describe('When user clicks on `Card`', () => {
		it('should show the answer', () => {
			component.find('.card').simulate('click');
			expect(component.find('.text-revealed').exists()).toBeTruthy();
		});
	});
});
