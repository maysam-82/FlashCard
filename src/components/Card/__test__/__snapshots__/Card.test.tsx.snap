// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render Card correclty 1`] = `
<div
  className="card border rounded m-2 p-2 d-flex  flex-row justify-content-between align-items-center"
  onClick={[Function]}
>
  <div
    className="card-prompt text-left"
  >
    <h4
      className="m-0"
    >
      test prompt
    </h4>
  </div>
  <div
    className="card-answer text-right"
  >
    <h4
      className="m-0 text-hidden"
    >
      test answer
    </h4>
  </div>
</div>
`;
